### PR TITLE
CORDA-3124: Ensure transaction is fully verified before sending to no…

### PR DESCRIPTION
**Pull request by adagys**
…tary

NotaryFlow.Client flow assumes the passed transaction is verified, which is
the case when it's invoked as a sub-flow in FinalityFlow. However,
NotaryFlow.Client can also be called directly from any custom flow, so
we need to ensure it performs transaction verification to avoid accidentally
sending an invalid transaction to a non-validating notary.

### CODE REVIEW

**Review by thschroeter — DISMISSED** [review]
LGTM

**Comment by lockathan** [comment]
Please could we change the branch to be merged into from release/os/4.3 to release/os/4.4? I don't think this needs to go into 4.3 and I'd rather avoid it going in by accident.

**Comment by adagys** [comment]
@lockathan Well, the ticket says it does need to go into 4.3, but that can be debated

**Comment by adagys** [comment]
It's technically a security fix by the way

**Comment by lockathan** [comment]
Happy for it to go into 4.3 if it is a security fix. Any potential impact?

File: core/src/main/kotlin/net/corda/core/flows/NotaryFlow.kt [hunk 1]
```diff
@@ -35,9 +38,16 @@ class NotaryFlow {
     @InitiatingFlow
     open class Client(
             private val stx: SignedTransaction,
-            override val progressTracker: ProgressTracker
+            override val progressTracker: ProgressTracker,
+            /**
+             * Set to *true* if the [stx] has already been verified for signature and contract validity,
+             * to prevent re-verification.
+             */
+            private val skipVerification: Boolean = false
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/flows/NotaryFlow.kt:46]: This has changed the default behaviour, which is OK I think, but it should be noted in the changelog.

Comment (adagys) [reply core/src/main/kotlin/net/corda/core/flows/NotaryFlow.kt:46]: Good point. For the regular path via FinalityFlow there's no behaviour change though

Comment (adagys) [reply core/src/main/kotlin/net/corda/core/flows/NotaryFlow.kt:46]: Done

**Review by shamsasari — DISMISSED** [review]

**Review by thschroeter — APPROVED** [review]
LGTM

**Pull request merged**